operator/src/main/java/oracle/kubernetes/operator/DomainStatusUpdater.java

```diff
@@ -173,12 +176,42 @@ public FailureCountStep(V1Job domainIntrospectorJob) {
     @Override
     void modifyStatus(DomainStatus domainStatus) {
       domainStatus.incrementIntrospectJobFailureCount(getJobUid());
+      addRetryInfoToStatusMessage(domainStatus);
     }
 
     @Nullable
     private String getJobUid() {
       return Optional.ofNullable(domainIntrospectorJob).map(V1Job::getMetadata).map(V1ObjectMeta::getUid).orElse(null);
     }
+
+    private void addRetryInfoToStatusMessage(DomainStatus domainStatus) {
+
+      if (hasExceededMaxRetryCount(domainStatus)) {
+        domainStatus.setMessage(createStatusMessage(domainStatus, EXCEEDED_INTROSPECTOR_MAX_RETRY_COUNT_ERROR_MSG));
+      } else if (isFatalError(domainStatus)) {
+        domainStatus.setMessage(createStatusMessage(domainStatus, FATAL_ERROR_DOMAIN_STATUS_MESSAGE));
+      } else {
+        domainStatus.setMessage(createStatusMessage(domainStatus, getNonFatalRetryStatusMessage(domainStatus)));
+      }
+    }
+
+    private boolean hasExceededMaxRetryCount(DomainStatus domainStatus) {
+      return domainStatus.getIntrospectJobFailureCount() >= DomainPresence.getDomainPresenceFailureRetryMaxCount();
+    }
+
+    private String createStatusMessage(DomainStatus domainStatus, String retryStatusMessage) {
+      return onSeparateLines(retryStatusMessage, INTROSPECTION_ERROR, domainStatus.getMessage());
+    }
+
+    private String getNonFatalRetryStatusMessage(DomainStatus domainStatus) {
+      return "Introspection failed on try " + domainStatus.getIntrospectJobFailureCount()
+              + " of " + DomainPresence.getDomainPresenceFailureRetryMaxCount() + ".";
```

Comment:
Shall we also provide the information about when the next retry will happen?

Reply:
@rjeberhard @tbarnes-us @jshum2479 - Any thoughts about this and where do we get this information from? Thanks.

Reply:
Would DomainPresence.getDomainPresenceFailureRetrySeconds() work for this?

Reply:
Do we need exact timing?  or can we just tell the user it will retry approximately in next make right interval (2 mins or the configured time).

Reply:
In some cases such as when an introspector job times out, we retry after `DomainPresence.getDomainPresenceFailureRetrySeconds` but in other cases we retry during next make right interval.

Reply:
We will track this new requirement with a separate JIRA and address in a new pull request.

Reply:
please let me know the retry JIRA# when filed

note there's an existing umbrella retry JIRA OWLS-84438 that's meant for brain-storming purposes.  

IMO, it's confusing to have two different retry settings - as this makes it hard for customers to figure out which setting they need to tune

Reply:
@tbarnes-us new JIRA is https://jira.oraclecorp.com/jira/browse/OWLS-93899 .